source/globalCommands.py

```diff
@@ -3260,6 +3264,31 @@ def script_braille_cycleCursorShape(self, gesture):
 		# Translators: Reports which braille cursor shape is activated.
 		ui.message(_("Braille cursor %s") % shapeMsg)
 
+	@script(
+		# Translators: Input help mode message for cycle through braille show messages command.
+		description=_("Cycle through the braille show messages modes"),
+		category=SCRCAT_BRAILLE
+	)
+	def script_braille_cycleShowMessages(self, gesture: inputCore.InputGesture) -> None:
+		"""Set next state of braille show messages and reports it with ui.message."""
+		values = [x.value for x in ShowMessages]
+		index = values.index(
+			config.conf["braille"]["showMessages"]
+		)
+		newIndex = (index + 1) % len(values)
+		newValue = values[newIndex]
+		config.conf["braille"]["showMessages"] = newValue
+		ui.message(
+			_(
+				# Translators: Reports which show braille message mode is used
+				# (disabled, timeout or indefinitely).
+				"Braille show messages %s"
+			)
+			% ShowMessages(
+				newValue
+			).displayString
+		)

```

Comment:
I think the whitespace could really be reduced here and readability improved

```suggestion
			# Translators: Reports which show braille message mode is used
			# (disabled, timeout or indefinitely).
			msg = _("Braille show messages %s") % ShowMessages(newValue).displayString
			ui.message(msg)
```